### PR TITLE
Replace range(len(foo)) with enumerate(foo).

### PR DESCRIPTION
**Pull request by rodrigc**
This eliminates some pylint warnings on Python 3.

This is a subset of the changes in #4470

### CODE REVIEW

File: master/buildbot/data/base.py [hunk 1]
```diff
@@ -53,8 +52,7 @@ def compileEventPathPatterns(self):
 
     def getEndpoints(self):
         endpoints = self.endpoints[:]
-        for i in range(len(endpoints)):
-            ep = endpoints[i]
+        for (i, ep) in enumerate(endpoints):
```
Comment (yan12125) [inline master/buildbot/data/base.py]: I think parentheses around `i, ep` are usually omitted? That's what I see in other Python projects, and that's also how `enumerate()` is used in [PEP 279](https://www.python.org/dev/peps/pep-0279/), the PEP proposing `enumerate()`.

Comment (rodrigc) [reply master/buildbot/data/base.py]: OK I got rid of the parentheses next to the enumerate() statements.

**Comment by seankelly** [comment]
I see a number of instances where `foo[i]` isn't replaced with the new `val`.

**Comment by tardyp** [comment]
> I see a number of instances where foo[i] isn't replaced with the new val.


indeed. We need to be careful, because the codecov for this code is not very good.

**Comment by rodrigc** [comment]
@tardyp I didn't use `val` in all the places, because in some places they do some things where they really need to get at the index variable `i`, so just left it like that

**Comment by p12tic** [comment]
I would like to see `foo[i]` to be replaced with `val` where possible too. Otherwise LGTM.

**Pull request merged**